library/core/src/main/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelector.java

```diff
@@ -1762,29 +1778,105 @@ private Format getVideoSelectedFormat(
+  private void updateOverride(int rendererIndex, Format selectedVideoFormat, ExoTrackSelection.Definition definition, MappedTrackInfo mappedTrackInfo, Parameters params) {
     if (definition == null) {
       return;
     }
+
+    // May update track type overrides.
+    SparseArray<Pair<TrackSelectionOverride, Integer>> applicableOverridesByTrackType = getApplicableOverrides(mappedTrackInfo, params);
+    int trackType = mappedTrackInfo.getRendererType(rendererIndex);
+    Pair<TrackSelectionOverride, Integer> existingOverride = applicableOverridesByTrackType.get(trackType);
+    TrackSelectionOverride existingTrackOverride = (existingOverride == null ? null : existingOverride.first);
+    if (existingTrackOverride != null) {
+      TrackGroup overrideGroup = existingTrackOverride.trackGroup;
+      int[] overrideTracks = Ints.toArray(existingTrackOverride.trackIndices);
+      if (mayUpdateOverride(selectedVideoFormat, trackType, overrideGroup, overrideTracks)) {
+        if (overrideGroup != definition.group || !Arrays.equals(overrideTracks, definition.tracks)) {
+          Parameters oldParameters = parametersReference.get();
+          TrackSelectionOverrides trackSelectionOverrides = oldParameters.trackSelectionOverrides.buildUpon()
+              .setOverrideForType(new TrackSelectionOverride(definition.group, Ints.asList(definition.tracks)))
+              .build();
+          parametersReference.set(oldParameters.buildUpon().setTrackSelectionOverrides(trackSelectionOverrides).build());
+        }
+      }
+    }
+
+    // May update legacy renderer overrides.
     TrackGroupArray trackGroups = mappedTrackInfo.getTrackGroups(rendererIndex);
     SelectionOverride override = params.getSelectionOverride(rendererIndex, trackGroups);
     if (override != null) {
-      int groupIndex = trackGroups.indexOf(definition.group);
-      int[] tracks = definition.tracks;
-      if (override.groupIndex != groupIndex || !Arrays.equals(override.tracks, tracks)) {
-        Map<TrackGroupArray, SelectionOverride> overrides = params.selectionOverrides.get(rendererIndex);
-        overrides.put(trackGroups, new SelectionOverride(groupIndex, tracks));
+      TrackGroup overrideGroup = trackGroups.get(override.groupIndex);
+      int[] overrideTracks = override.tracks;
+      if (mayUpdateOverride(selectedVideoFormat, trackType, overrideGroup, overrideTracks)) {
+        int groupIndex = trackGroups.indexOf(definition.group);
+        int[] tracks = definition.tracks;
+        if (override.groupIndex != groupIndex || !Arrays.equals(override.tracks, tracks)) {
+          Map<TrackGroupArray, SelectionOverride> overrides = params.selectionOverrides.get(rendererIndex);
+          overrides.put(trackGroups, new SelectionOverride(groupIndex, tracks));
+        }
       }
     }
   }
 
+  private boolean mayUpdateOverride(Format selectedVideoFormat, int trackType, TrackGroup overrideGroup, int[] overrideTracks) {
```

Comment:
nit - I would rename this to `shouldUpdateOverride`

Reply:
changed.